piff/gp_interp.py

```
@@ -0,0 +1,324 @@
+        from numpy import array
+        return eval(kernel)
+
+    def _fit(self, X, y, logger=None):
+        """Update the GaussianProcessRegressor with data
+        :param X:  The independent covariates.  (n_samples, n_features)
+        :param y:  The dependent responses.  (n_samples, n_targets)
+        """
+        # Save these for potential read/write.
+        self._X = X
+        self._y = y
+        if self.npca > 0:
+            from sklearn.decomposition import PCA
+            self._pca = PCA(n_components=self.npca, whiten=True)
+            self._pca.fit(y)
+            y = self._pca.transform(y)
+        self.gp.fit(X, y)
+
+    def _predict(self, Xstar):
+        """ Predict responses given covariates.
+        :param X:  The independent covariates at which to interpolate.  (n_samples, n_features).
+        :returns:  Regressed parameters  (n_samples, n_targets)
+        """
+        ystar = self.gp.predict(Xstar)
+        if self.npca > 0:
+            ystar = self._pca.inverse_transform(ystar)
+        return ystar
+
+    def getProperties(self, star, logger=None):
+        """Extract the appropriate properties to use as the independent variables for the
+        interpolation.
+
+        Take self.attr_interp from star.data
+
+        :param star:    A Star instances from which to extract the properties to use.
+
+        :returns:       A np vector of these properties.
+        """
+        return np.array([star.data[key] for key in self.attr_interp])
+
+    def initialize(self, stars, logger=None):
+        """Initialize both the interpolator to some state prefatory to any solve iterations and
+        initialize the stars for use with this interpolator.
+
+        :param stars:   A list of Star instances to interpolate between
+        :param logger:  A logger object for logging debug info. [default: None]
+        """
+        self.solve(stars, logger=logger)
+        return self.interpolateList(stars)
```

Does this mean that for every interp, we need to be able to catch if the solving has not occurred? It seems weird to me to have a silent failure for interpolating with an interpolant that hasn't solved...